components/lane_follower/src/individual_follower.py

```diff
@@ -0,0 +1,172 @@
+                return None, 0, 0, 0
+            lane_array = np.concatenate(lane_inds)
+            lane_x_pos = nonzero_x[lane_array]
+            lane_y_pos = nonzero_y[lane_array]
+
+            # I don't believe this if statement is necessary, need to test
+            if lane_x_pos.any() and lane_y_pos.any():
+                self._fit = np.polyfit(
+                    lane_y_pos, lane_x_pos, IndividualFollower.LANE_POLY_SIZE
+                )
+
+            plotting_coordinates = np.linspace(
+                0,
+                self._binary_warped.shape[0] - 1,
+                self._binary_warped.shape[1],
+            )
+
+            polynomial = np.polyval(self._fit, plotting_coordinates)
+            out_img[nonzero_y[lane_array], nonzero_x[lane_array]] = (
+                IndividualFollower.LANE_COLOR
+            )
+
+            ##Generates the search window area
+            window_img = np.zeros_like(out_img)
+            ##These lines should be broken up accordingly: They render the search area
+            line_window1 = np.array(
+                [
+                    np.transpose(
+                        np.vstack(
+                            [
+                                polynomial - IndividualFollower.SEARCH_WIDTH,
+                                plotting_coordinates,
+                            ]
+                        )
+                    )
+                ]
+            )
+        line_window2 = np.array(
+            [
+                np.transpose(
+                    np.vstack(
+                        [
+                            polynomial + IndividualFollower.SEARCH_WIDTH,
+                            plotting_coordinates,
+                        ]
+                    )
+                )
+            ]
+        )
+        line_pts = np.hstack((line_window1, line_window2))
+        cv2.fillPoly(
+            window_img, np.int_([line_pts]), color=IndividualFollower.BOX_COLOR
+        )
+        line_pts = np.array(
+            [np.transpose(np.vstack([polynomial, plotting_coordinates]))],
+            dtype=np.int32,
+        )
+        cv2.polylines(
+            out_img,
+            line_pts,
+            isClosed=False,
+            color=IndividualFollower.LANE_COLOR,
+            thickness=IndividualFollower.DRAW_THICKNESS,
+        )
+
+        # Calculating heading error by converting lane polynomial into line
+        ##TODO: Make this use the furthest found box. This way, we'll use the most accurate heading
+        y_lower = 0
+        y_upper = (IndividualFollower.NWINDOWS + 1) * window_height
+        slope = np.polyval(self.fit, y_lower) - np.polyval(
+            self.fit, y_upper
+        ) / (y_upper - y_lower)
+        heading = math.atan(slope)
+        result = cv2.addWeighted(out_img, 1, window_img, 0.3, 0)
+        # TODO: Determine what result really is, and annotate efficiently
+        return result, empty_windows, heading, slope
```

Comment:
It's better to return some kind of result object (maybe a named tuple) so that it's easy to unpack results.